<a href="https://colab.research.google.com/github/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/Pre-Entrega_Maune%2C%20Juan_Federico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


<img  src="https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/Pre-Entrega%20IMG.PNG" width=100%>

#**MAUNE, Juan Federico**
#**Pre-Entrega de Proyecto**

**Bases de datos a utilizar:**
- [clientes.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/clientes.csv)
- [marketing.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/marketing.csv)
- [ventas.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/ventas.csv)


#**Etapa 1: Recopilación y preparación de Datos**

##***Actividad 1: Carga de los sets de datos***

In [3]:
# se importa pandas
import pandas as pd

In [4]:
# se importa data set clientes
df_clientes = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/clientes.csv")

In [5]:
# se importa data set marketing
df_marketing = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/marketing.csv")

In [6]:
# se importa data set ventas
df_ventas = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/ventas.csv")

##***Actividad 2 y 3: Script Básico***

In [7]:
# se paso parte de los registros del set de datos de "Ventas" a una lista de diccionarios
lista_ventas = [
{'id_venta': 2824, 'producto': 'Tablet', 'precio': 74.09, 'cantidad': 5, 'fecha_venta': '45294', 'categoria': 'Electrónica'},
{'id_venta': 446, 'producto': 'Cortinas', 'precio': 62.10, 'cantidad': 4, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 630, 'producto': 'Adorno de pared', 'precio': 71.99, 'cantidad': 9, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 794, 'producto': 'Jarrón decorativo', 'precio': 96.97, 'cantidad': 7, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 882, 'producto': 'Cuadro decorativo', 'precio': 101.92, 'cantidad': 9, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 969, 'producto': 'Candelabro', 'precio': 90.70, 'cantidad': 12, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 1445, 'producto': 'Lavadora', 'precio': 28.41, 'cantidad': 10, 'fecha_venta': '45295', 'categoria': 'Electrodomésticos'},
{'id_venta': 1971, 'producto': 'Freidora eléctrica', 'precio': 122.25, 'cantidad': 11, 'fecha_venta': '45295', 'categoria': 'Electrodomésticos'},
{'id_venta': 2209, 'producto': 'Proyector', 'precio': 32.15, 'cantidad': 1, 'fecha_venta': '45295', 'categoria': 'Electrónica'},
{'id_venta': 256, 'producto': 'Cortinas', 'precio': 86.82, 'cantidad': 8, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 606, 'producto': 'Cortinas', 'precio': 106.66, 'cantidad': 11, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 704, 'producto': 'Jarrón decorativo', 'precio': 57.22, 'cantidad': 9, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 1362, 'producto': 'Heladera', 'precio': 109.49, 'cantidad': 11, 'fecha_venta': '45296', 'categoria': 'Electrodomésticos'},
{'id_venta': 2155, 'producto': 'Auriculares', 'precio': 84.80, 'cantidad': 10, 'fecha_venta': '45296', 'categoria': 'Electrónica'},
{'id_venta': 2947, 'producto': 'Cámara digital', 'precio': 107.78, 'cantidad': 3, 'fecha_venta': '45296', 'categoria': 'Electrónica'},
{'id_venta': 1495, 'producto': 'Lavadora', 'precio': 38.99, 'cantidad': 6, 'fecha_venta': '45297', 'categoria': 'Electrodomésticos'},
{'id_venta': 1572, 'producto': 'Cafetera', 'precio': 61.40, 'cantidad': 9, 'fecha_venta': '45297', 'categoria': 'Electrodomésticos'},
{'id_venta': 2180, 'producto': 'Consola de videojuegos', 'precio': 102.57, 'cantidad': 9, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2487, 'producto': 'Cámara digital', 'precio': 50.76, 'cantidad': 3, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2603, 'producto': 'Smartphone', 'precio': 87.05, 'cantidad': 5, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2885, 'producto': 'Auriculares', 'precio': 100.40, 'cantidad': 6, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 17, 'producto': 'Elementos de cerámica', 'precio': 108.48, 'cantidad': 11, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 183, 'producto': 'Espejo decorativo', 'precio': 82.76, 'cantidad': 8, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 560, 'producto': 'Adorno de pared', 'precio': 57.62, 'cantidad': 6, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 632, 'producto': 'Cuadro decorativo', 'precio': 30.49, 'cantidad': 12, 'fecha_venta': '45298', 'categoria': 'Decoración'}
]

In [8]:
# ventas acumuladas de la categoría "Decoración"
ventas_decoracion = sum( [venta ['precio']* venta['cantidad']
                          for venta in lista_ventas
                          if venta['categoria']=='Decoración'])
print(f'El monto obtenido por las ventas en la categoría Decoración es de $ {ventas_decoracion}')

El monto obtenido por las ventas en la categoría Decoración es de $ 8530.539999999999


In [9]:
# ventas acumuladas de la categoría "Electrodométicos"
ventas_electrodomesticos = sum( [venta ['precio']* venta['cantidad']
                                 for venta in lista_ventas
                                 if venta['categoria']=='Electrodomésticos'])
print(f'El monto obtenido por las ventas en la categoría Electrodomésticos es de $ {ventas_electrodomesticos}')

El monto obtenido por las ventas en la categoría Electrodomésticos es de $ 3619.7799999999997


In [10]:
# ventas acumuladas de la categoría "Electrónica"
ventas_electronica = sum( [venta ['precio']* venta['cantidad']
                           for venta in lista_ventas
                           if venta['categoria']=='Electrónica'])
print(f'El monto obtenido por las ventas en la categoría Electrónica es de $ {ventas_electronica}')

El monto obtenido por las ventas en la categoría Electrónica es de $ 3687.0


In [11]:
# utilicé la AI para obtener un código que unifique los codigos anteriores en una sola sentencia
ventas_por_categoria = {
    categoria: sum(venta['precio'] * venta['cantidad']
                   for venta in lista_ventas if venta['categoria'] == categoria)
    for categoria in {venta['categoria'] for venta in lista_ventas}
}

print(ventas_por_categoria)

{'Electrodomésticos': 3619.7799999999997, 'Electrónica': 3687.0, 'Decoración': 8530.539999999999}


## ***Actividad 4: Calidad de los Datos***
**Análisis exploratorio con Pandas**

### ***DataFrame Clientes***

#### **Análisis Exploratorio**

In [12]:
# Realizo un paneo del dataset mediante info() para conocer columnas, filas, tipos de datos y si hay datos nulos
df_clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id_cliente  567 non-null    int64  
 1   nombre      567 non-null    object 
 2   edad        567 non-null    int64  
 3   ciudad      567 non-null    object 
 4   ingresos    567 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 22.3+ KB


In [13]:
# obtengo una muestra medinte sample() para ver los datos en la tabla y ver si hay coherencia en la misma.
df_clientes.sample(10)

id_cliente             nombre  edad                 ciudad  ingresos
499         500      Gunar Antonio    40  San Miguel de Tucumán  58970.48
249         250     Danita Bensley    43                  Salta  28001.09
352         353         Camey Cloy    54                Rosario  27324.28
149         150      Jaclyn Crepin    56             Corrientes  23760.91
427         428  Jessee Birmingham    40            Resistencia  27450.93
534         535    Clarey Roglieri    27                  Merlo  25404.19
181         182      Baldwin Drieu    39                  Salta  23533.08
209         210        Stephi Ties    38           Bahía Blanca  39717.60
461         462     Cherilynn Inch    29                  Salta  19754.23
340         341        Karena Athy    31               Santa Fe   8996.96

In [14]:
# utilizo describe() para obtener estadísticas de las columnas numéricas y corroborar los rangos de los datos
df_clientes.describe()

id_cliente        edad      ingresos
count  567.000000  567.000000    567.000000
mean   284.000000   37.940035  34668.739012
std    163.823075   10.202885  12974.531446
min      1.000000   20.000000    170.290000
25%    142.500000   30.000000  26015.240000
50%    284.000000   37.000000  35066.830000
75%    425.500000   43.000000  42457.100000
max    567.000000   81.000000  88053.010000

In [15]:
# utilizo value_count() sobre la columna ciudad para ver si hay ciudades cargadas de diferente forma
df_clientes['ciudad'].value_counts()

ciudad
Mar del Plata            63
Rosario                  55
Posadas                  52
Resistencia              50
Córdoba                  49
Corrientes               47
Santa Fe                 46
Bahía Blanca             44
Salta                    43
Merlo                    43
San Miguel de Tucumán    39
Buenos Aires             36
Name: count, dtype: int64

In [16]:
# ordenos los nombres en orden ascendente para ver si hay nombres que inician con números o simbolos
df_clientes['nombre'].sort_values(ascending=True).head(20)

519          Aarika Brehat
403             Ab Wilsher
349        Abbey Vanichkin
184            Addy Lattos
103        Adolphus Morton
333         Adrianna Fenby
96                Ag Ifill
67         Agustin Bofield
536          Aidan Leavens
327        Ailina Martelet
70          Aindrea Winter
361    Albrecht Itchingham
245         Aldous Stivers
458     Alethea Jacobovitz
237      Alexandra Stapley
544             Ali Roxbee
529            Alicea Clow
63          Alison Ianizzi
148       Allissa Napleton
0          Aloysia Screase
Name: nombre, dtype: object

In [17]:
# use AI para obtener una sentencia que verifique la presencia de números o simbolos en la columna nombres
df_clientes[df_clientes["nombre"].str.contains(r"[^a-zA-ZáéíóúÁÉÍÓÚñÑ\s]", na=False)]

#Debio a que en la salida aprarecienron nombres ingles que contienen guien medio(-), apóstrofe (') y puntos (.), las entencia se cambió a :

#df_clientes[df_clientes["nombre"].str.contains(r"[^a-zA-ZáéíóúÁÉÍÓÚñÑ\s\.\-']", na=False)]

id_cliente                nombre  edad                 ciudad  ingresos
29           30          Tedi O'Shiel    32           Bahía Blanca  44137.90
124         125        Bennie O'Regan    40            Resistencia  60645.26
153         154    My de'-Ancy Willis    43                Rosario  33105.01
156         157   Westbrook O'Caherny    25               Santa Fe  29982.55
159         160    Holly-anne Whymark    43          Mar del Plata  30588.42
187         188          D'arcy Creak    42           Buenos Aires  36868.84
246         247      Shaw Dell'Abbate    33                Posadas  41806.83
268         269      Carilyn O'Mailey    52           Bahía Blanca   3603.65
290         291  Marie-jeanne Johnson    28               Santa Fe  42821.99
294         295   Bette-ann Bertholin    40           Bahía Blanca  35319.17
417         418   Melita Watson-Brown    27  San Miguel de Tucumán  22533.03
445         446       Gillan St. Hill    29          Mar del Plata  52757.36
466         467     Felipe O'Hallagan    31             Corrientes  32928.99
472         473         Dollie O'Roan    25            Resistencia  51350.49
545         546     Roseline L'oiseau    39             Corrientes  44987.46

In [18]:
# corroboro la auscencia de valores nulos
df_clientes.isnull().sum()

id_cliente    0
nombre        0
edad          0
ciudad        0
ingresos      0
dtype: int64

In [19]:
# exploro la presencia de registros duplicados
df_clientes.duplicated().sum()

np.int64(0)

####**Observaciones**


- cantidad de registros: 567
- columnas: 5
- Precisión:
    - los datos estan en los rangos esperados
- Validez:
    - los datos poseen los tipos de datos correctos
    - los nombres no poseen números ni símbolos
- Completitud:
    - no se observan registros con valores nulos
- Consistencia:
    - las ciudades están escritas de igual forma
    - no hay registros duplicados
- **Cambios a realizar:**
    - no se obserban

### ***DataFrame Marketing***

#### **Análisis Exploratorio**

In [20]:
# Realizo un paneo del dataset mediante info() para conocer columnas, filas, tipos de datos y si hay datos nulos
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id_campanha   90 non-null     int64  
 1   producto      90 non-null     object 
 2   canal         90 non-null     object 
 3   costo         90 non-null     float64
 4   fecha_inicio  90 non-null     object 
 5   fecha_fin     90 non-null     object 
dtypes: float64(1), int64(1), object(4)
memory usage: 4.3+ KB


In [21]:
# obtengo una muestra medinte sample() para ver los datos en la tabla y ver si hay coherencia en la misma.
df_marketing.sample(10)

id_campanha            producto  canal  costo fecha_inicio   fecha_fin
55           62     Lámpara de mesa     TV   4.51   26/08/2024  07/10/2024
48           79            Cafetera     TV   7.26   13/08/2024  11/10/2024
26           71  Freidora eléctrica     TV   4.37   05/06/2024  24/07/2024
19           51          Smartphone  Email   4.65   28/05/2024  15/06/2024
69           55          SmartWatch  Email   4.59   13/10/2024   6/12/2024
27           38   Rincón de plantas  Email   5.76   14/06/2024  15/07/2024
39           53   Espejo decorativo  Email   4.90   17/07/2024  01/08/2024
86           89           Televisor     TV   4.98   13/12/2024    8/2/2025
1            12              Tablet   RRSS   3.40   26/03/2024  13/05/2024
38           39          Candelabro  Email   6.08   17/07/2024  18/08/2024

In [22]:
# utilizo describe() para obtener estadísticas de las columnas numéricas y corroborar los rangos de los datos
df_marketing.describe()

id_campanha      costo
count    90.000000  90.000000
mean     45.500000   4.928667
std      26.124701   0.947750
min       1.000000   2.950000
25%      23.250000   4.372500
50%      45.500000   4.900000
75%      67.750000   5.562500
max      90.000000   7.390000

In [23]:
# utilizo value_count() sobre la columna "producto" para ver si hay un mismo producto está cargado con diferentes nombres
df_marketing['producto'].sort_values().value_counts()

producto
Adorno de pared           3
Alfombra                  3
Aspiradora                3
Auriculares               3
Batidora                  3
Cafetera                  3
Candelabro                3
Consola de videojuegos    3
Cortinas                  3
Cuadro decorativo         3
Cámara digital            3
Elementos de cerámica     3
Espejo decorativo         3
Freidora eléctrica        3
Heladera                  3
Horno eléctrico           3
Jarrón decorativo         3
Laptop                    3
Lavadora                  3
Lámpara de mesa           3
Microondas                3
Parlantes Bluetooth       3
Plancha de vapor          3
Proyector                 3
Rincón de plantas         3
Secadora                  3
SmartWatch                3
Smartphone                3
Tablet                    3
Televisor                 3
Name: count, dtype: int64

In [24]:
# utilizo value_count() sobre la columna "canal" para ver si un canal está cargado con diferentes nombres
df_marketing['canal'].sort_values().value_counts()

canal
Email    30
RRSS     30
TV       30
Name: count, dtype: int64

In [25]:
# corroboro la auscencia de valores nulos
df_marketing.isnull().sum()

id_campanha     0
producto        0
canal           0
costo           0
fecha_inicio    0
fecha_fin       0
dtype: int64

In [26]:
# exploro la presencia de registros duplicados
df_marketing.duplicated().sum()

np.int64(0)

####**Observaciones**

- cantidad de registros: 90
- columnas: 6
- Precisión:
    - los datos estan en los rangos esperados
    - los registro de fechas no pueden ser analizado
- Validez:
    - las columnas 'fecha_inicio' y 'fecha_fin' tiene con Dtype Object y están en formato *dd/mm/aaaa*
- Completitud:
    - no se observan registros con valores nulos
- Consistencia:
    - las productos y canales están escritas de igual forma
    - no hay registros duplicados
- **Cambios a realizar:**
    1. cambiar el Dtype de 'fecha_inicio' y 'fecha_fin' a *datetime*

### ***DataFrame Ventas***

#### **Análisis Exploratorio**

In [27]:
# Realizo un paneo del dataset mediante info() para conocer columnas, filas, tipos de datos y si hay datos nulos
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3035 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     3035 non-null   int64  
 1   producto     3035 non-null   object 
 2   precio       3033 non-null   object 
 3   cantidad     3033 non-null   float64
 4   fecha_venta  3035 non-null   object 
 5   categoria    3035 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 142.4+ KB


In [28]:
# obtengo una muestra medinte sample() para ver los datos en la tabla y ver si hay coherencia en la misma.
df_ventas.sample(10)

id_venta                producto   precio  cantidad fecha_venta  \
1142      2898              SmartWatch  $108.81      10.0  05/05/2024   
2559       582       Cuadro decorativo   $61.67      10.0  01/11/2024   
1914      1289                Batidora   $76.71       7.0  11/08/2024   
1575      1826                Secadora  $109.43       9.0  29/06/2024   
566         23       Espejo decorativo   $70.36       5.0  08/03/2024   
841       2080  Consola de videojuegos  $100.69       7.0  06/04/2024   
415       1115                Lavadora   $96.75       3.0  18/02/2024   
1301       859         Lámpara de mesa   $76.25       4.0  23/05/2024   
2986       120         Adorno de pared   $64.26       6.0  25/12/2024   
2722       837   Elementos de cerámica   $56.10      12.0  22/11/2024   

              categoria  
1142        Electrónica  
2559         Decoración  
1914  Electrodomésticos  
1575  Electrodomésticos  
566          Decoración  
841         Electrónica  
415   Electrodomésticos  
1301         Decoración  
2986         Decoración  
2722         Decoración

In [29]:
# utilizo describe() para obtener estadísticas de las columnas numéricas y corroborar los rangos de los datos
df_ventas.describe()

id_venta     cantidad
count  3035.000000  3033.000000
mean   1499.851400     6.496538
std     866.465379     3.457250
min       1.000000     1.000000
25%     748.500000     3.000000
50%    1502.000000     7.000000
75%    2249.500000     9.000000
max    3000.000000    12.000000

In [30]:
# utilizo value_count() sobre la columna "productos" para ver si un producto está cargado con diferentes nombres
df_ventas['producto'].value_counts()

producto
Lámpara de mesa           181
Auriculares               147
Microondas                135
Cafetera                  118
Laptop                    104
Secadora                  103
Jarrón decorativo         102
Smartphone                102
Alfombra                  102
SmartWatch                102
Aspiradora                102
Cortinas                  102
Rincón de plantas         102
Freidora eléctrica        101
Espejo decorativo         101
Adorno de pared           101
Lavadora                  100
Consola de videojuegos    100
Cuadro decorativo         100
Horno eléctrico           100
Plancha de vapor          100
Proyector                 100
Elementos de cerámica     100
Batidora                  100
Televisor                 100
Cámara digital            100
Parlantes Bluetooth        99
Tablet                     58
Heladera                   49
Candelabro                 24
Name: count, dtype: int64

In [31]:
# utilizo value_count() sobre la columna "categoria" para ver si una categoria esta cargada con diferentes nombres
df_ventas['categoria'].value_counts()

categoria
Decoración           1015
Electrónica          1012
Electrodomésticos    1008
Name: count, dtype: int64

In [32]:
# corroboro la auscencia de valores nulos
df_ventas.isnull().sum()

id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

In [33]:
# visualizo los volares nulos
df_ventas[df_ventas.isnull().any(axis=1)]

id_venta               producto precio  cantidad fecha_venta    categoria
136       627  Elementos de cerámica    NaN       NaN  17/01/2024   Decoración
139      2171    Parlantes Bluetooth    NaN       NaN  17/01/2024  Electrónica

In [34]:
# exploro la presencia de registros duplicados
df_ventas.duplicated().sum()

np.int64(35)

In [35]:
# visualizo los valores duplicados
df_ventas[df_ventas.duplicated(keep=False)].sort_values(by="id_venta")

id_venta           producto  precio  cantidad fecha_venta    categoria
820        56           Cortinas  $66.24       5.0  05/04/2024   Decoración
826        56           Cortinas  $66.24       5.0  05/04/2024   Decoración
832       145           Alfombra  $41.72       8.0  06/04/2024   Decoración
844       145           Alfombra  $41.72       8.0  06/04/2024   Decoración
878       204  Jarrón decorativo  $94.75       1.0  08/04/2024   Decoración
..        ...                ...     ...       ...         ...          ...
830      2545        Auriculares  $32.81      11.0  05/04/2024  Electrónica
883      2705        Auriculares  $87.29       2.0  08/04/2024  Electrónica
889      2705        Auriculares  $87.29       2.0  08/04/2024  Electrónica
831      2778         SmartWatch  $70.66       7.0  05/04/2024  Electrónica
825      2778         SmartWatch  $70.66       7.0  05/04/2024  Electrónica

[70 rows x 6 columns]

####**Observaciones**


precio (object) --> float64

fecha_venta (object) --> datetime

cantidad (float64) --> int64

los precios inclyen el simbolo ($), hay que quitarlos

no obsevo valores nulos con info, hay 2 registros con valres nulos  (0.07% de los registros)

hay 35 valores duplicados

- cantidad de registros: 3035
- columnas: 6
- Precisión:
    - los datos de *'cantidad'* están en rangos esperables. No hay valores negativos
    - no se puede indicar la precisión de *'precio'*, *'fecha_venta'* por tener el Dtype object
- Validez:
    - las columnas *'precio'* y *'fecha_venta'* poseen el Dtype Object. *'precio'* posee el prefijo '$'. *'fecha_venta'* está en formato *dd/mm/aaaa*
- Completitud:
    - se observa 2 registros con valores nulos (2/3035 => 0.07% de los registros)
- Consistencia:
    - los valores de categoría y productos están escritos de igual forma
    - hay 35 registros duplicados
- **Cambios a realizar:**
    1. eliminar los 2 registros con datos nulos
    2. eliminar los registros duplicados
    3. eliminar el prefijo '$' de los datos de *'precio'*
    4. cambiar Dtype de *'precio'* a *float64*
    5. cambiar Dtype de *'fecha_venta'* a *datetime*

#**Etapa 2: Preprocesamiento y Limpieza de Datos**

##***Actividad 1: Limpieza de Datos***

### ***DataFrame Marketing***

- **Cambios a realizar:**
    1. cambiar el Dtype de 'fecha_inicio' y 'fecha_fin' a *datetime*

In [36]:
# corroboro estado inicial del dataframe
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id_campanha   90 non-null     int64  
 1   producto      90 non-null     object 
 2   canal         90 non-null     object 
 3   costo         90 non-null     float64
 4   fecha_inicio  90 non-null     object 
 5   fecha_fin     90 non-null     object 
dtypes: float64(1), int64(1), object(4)
memory usage: 4.3+ KB


In [37]:
# se convierte el campo fecha_inicio a dtype datetime
df_marketing['fecha_inicio'] = pd.to_datetime(df_marketing['fecha_inicio'], format="%d/%m/%Y")

In [38]:
# se convierte el campo fecha_inicio a dtype datetime
df_marketing['fecha_fin'] = pd.to_datetime(df_marketing['fecha_fin'], format="%d/%m/%Y")

In [39]:
# corroboro los cambios de Dtype realizados
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_campanha   90 non-null     int64         
 1   producto      90 non-null     object        
 2   canal         90 non-null     object        
 3   costo         90 non-null     float64       
 4   fecha_inicio  90 non-null     datetime64[ns]
 5   fecha_fin     90 non-null     datetime64[ns]
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 4.3+ KB


### ***DataFrame Ventas***

- **Cambios a realizar:**
    1. eliminar los 2 registros con datos nulos
    2. eliminar los registros duplicados
    3. eliminar el prefijo '$' de los datos de *'precio'*
    4. cambiar Dtype de *'precio'* a *float64*
    5. cambiar Dtype de *'fecha_venta'* a *datetime*

####*1. Eliminación de registros con datos nulos*

In [40]:
# corroboro presencia de registros con valores null
df_ventas[df_ventas.isnull().any(axis=1)]

id_venta               producto precio  cantidad fecha_venta    categoria
136       627  Elementos de cerámica    NaN       NaN  17/01/2024   Decoración
139      2171    Parlantes Bluetooth    NaN       NaN  17/01/2024  Electrónica

In [41]:
# Elimino los registros con valores nulos. Aplico los cambios en el mismo DataFrame
df_ventas.dropna(inplace=True)

In [42]:
# corroboro la eliminación de los registros con valores null
df_ventas[df_ventas.isnull().any(axis=1)]

Empty DataFrame
Columns: [id_venta, producto, precio, cantidad, fecha_venta, categoria]
Index: []

In [43]:
# corroboro que tengo 2 registro menos en el DataFrame Ventas
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3033 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     3033 non-null   int64  
 1   producto     3033 non-null   object 
 2   precio       3033 non-null   object 
 3   cantidad     3033 non-null   float64
 4   fecha_venta  3033 non-null   object 
 5   categoria    3033 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 165.9+ KB


####*2. Eliminación de registros duplicados*

In [44]:
# corroboro la presencia de los registros duplicados ordanados por id_venta
df_ventas[df_ventas.duplicated(keep=False)].sort_values(by="id_venta")

id_venta           producto  precio  cantidad fecha_venta    categoria
820        56           Cortinas  $66.24       5.0  05/04/2024   Decoración
826        56           Cortinas  $66.24       5.0  05/04/2024   Decoración
832       145           Alfombra  $41.72       8.0  06/04/2024   Decoración
844       145           Alfombra  $41.72       8.0  06/04/2024   Decoración
878       204  Jarrón decorativo  $94.75       1.0  08/04/2024   Decoración
..        ...                ...     ...       ...         ...          ...
830      2545        Auriculares  $32.81      11.0  05/04/2024  Electrónica
883      2705        Auriculares  $87.29       2.0  08/04/2024  Electrónica
889      2705        Auriculares  $87.29       2.0  08/04/2024  Electrónica
831      2778         SmartWatch  $70.66       7.0  05/04/2024  Electrónica
825      2778         SmartWatch  $70.66       7.0  05/04/2024  Electrónica

[70 rows x 6 columns]

In [45]:
# elimino los registros duplicados, aplico los cambios en el mismo DataFrame y reinico el indice del DataFrame
df_ventas.drop_duplicates(inplace=True, ignore_index=True)

In [46]:
# corroboro la eliminacionde los registros duplicados
df_ventas[df_ventas.duplicated(keep=False)].sort_values(by="id_venta")

Empty DataFrame
Columns: [id_venta, producto, precio, cantidad, fecha_venta, categoria]
Index: []

In [47]:
# vuelvo a ver el número de registros en el DataFrame
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   object 
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 140.7+ KB


####*3. Eliminación el prefijo '$' de los datos de 'precio'*

In [48]:
# corroboro la presencia del prefijo '$' en la columna 'precio'
df_ventas.sample(10)

id_venta            producto  precio  cantidad fecha_venta  \
2589      2087      Cámara digital  $35.44       3.0  10/11/2024   
1505      1825            Lavadora  $93.71       1.0  25/06/2024   
1951       940     Adorno de pared  $67.34       4.0  21/08/2024   
192        164   Jarrón decorativo  $66.27       1.0  24/01/2024   
1286      1721  Freidora eléctrica  $26.06       5.0  26/05/2024   
1871       295            Alfombra  $38.83      10.0  10/08/2024   
2873      1492            Heladera  $88.58      10.0  15/12/2024   
2698       923   Espejo decorativo  $59.64       2.0  24/11/2024   
1356      1923          Microondas  $97.54       4.0  03/06/2024   
608       1248            Cafetera  $30.70       5.0  12/03/2024   

              categoria  
2589        Electrónica  
1505  Electrodomésticos  
1951         Decoración  
192          Decoración  
1286  Electrodomésticos  
1871         Decoración  
2873  Electrodomésticos  
2698         Decoración  
1356  Electrodomésticos  
608   Electrodomésticos

In [49]:
# Elimino los posibles espacios en blanco y Reeplazo el simbolo '$' por nada
df_ventas["precio"] = df_ventas["precio"].str.strip().str.replace("$", "")

In [50]:
# corroboro la ausencia del prefijo '$' en la columna 'precio'
df_ventas.sample(10)

id_venta           producto  precio  cantidad fecha_venta  \
1024      2465        Auriculares   93.52       3.0  27/04/2024   
1476       410    Adorno de pared  109.64       3.0  21/06/2024   
2492      2217     Cámara digital   49.71      11.0  28/10/2024   
1388       648  Rincón de plantas   78.47       6.0  08/06/2024   
975        741    Lámpara de mesa   98.00      12.0  22/04/2024   
1639      1427    Horno eléctrico   33.55      11.0  12/07/2024   
2411      1326           Secadora  109.40       7.0  18/10/2024   
2880        69    Lámpara de mesa   93.24      11.0  16/12/2024   
524       1089           Batidora   61.72       8.0  02/03/2024   
987        945           Alfombra   60.74      12.0  24/04/2024   

              categoria  
1024        Electrónica  
1476         Decoración  
2492        Electrónica  
1388         Decoración  
975          Decoración  
1639  Electrodomésticos  
2411  Electrodomésticos  
2880         Decoración  
524   Electrodomésticos  
987          Decoración

####*4. Modificación del Dtype de *'precio'* a *float64**

In [51]:
# corroboro estado inicial del dataframe
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   object 
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 140.7+ KB


In [52]:
# convierto los valore de la columna 'precio' al Dtype float64
df_ventas['precio'] = df_ventas['precio'].astype('float64')

In [53]:
# corroboro el cambio de Dtype de 'precio'
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   float64
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(2), int64(1), object(3)
memory usage: 140.7+ KB


####*5. Modificación del Dtype de *'fecha_venta'* a *datetime**

In [54]:
# corroboro estado inicial del dataframe
df_ventas.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   float64
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(2), int64(1), object(3)
memory usage: 140.7+ KB


In [55]:
# se convierte el campo fecha_venta a dtype datetime
df_ventas['fecha_venta'] = pd.to_datetime(df_ventas['fecha_venta'], format="%d/%m/%Y")

In [56]:
# corroboro el cambio de Dtype de 'fecha_venta'
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id_venta     2998 non-null   int64         
 1   producto     2998 non-null   object        
 2   precio       2998 non-null   float64       
 3   cantidad     2998 non-null   float64       
 4   fecha_venta  2998 non-null   datetime64[ns]
 5   categoria    2998 non-null   object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(2)
memory usage: 140.7+ KB


##***Actividad 2: Transformacion de Datos***

Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento

In [57]:
# se crea una nueva columna 'monto_venta' = 'precio'*'cantidad'
df_ventas['monto_venta'] = df_ventas['precio']*df_ventas['cantidad']

In [58]:
# visualizo la creación de la nuevo columna
df_ventas.sample(15)

id_venta            producto  precio  cantidad fecha_venta  \
1580      1941  Freidora eléctrica   72.66       2.0  2024-07-04   
1255       812   Cuadro decorativo  107.32      10.0  2024-05-22   
1769      2786              Laptop  123.95      11.0  2024-07-28   
846       2373          Smartphone  123.60      12.0  2024-04-07   
296         64   Jarrón decorativo  104.16      10.0  2024-02-04   
2977      2038          SmartWatch  100.58       5.0  2024-12-28   
34        2603          Smartphone   87.05       5.0  2024-01-06   
345        229     Lámpara de mesa   99.71       3.0  2024-02-10   
2370      1113          Microondas   79.45       3.0  2024-10-11   
527       2777      Cámara digital   31.52       8.0  2024-03-02   
1847      1312          Microondas  118.10      10.0  2024-08-06   
610       2282           Televisor   31.80      10.0  2024-03-12   
2721      1980    Plancha de vapor  115.37       2.0  2024-11-26   
2529       808   Rincón de plantas   67.82      11.0  2024-11-02   
2123      1573          Microondas  121.44       4.0  2024-09-10   

              categoria  monto_venta  
1580  Electrodomésticos       145.32  
1255         Decoración      1073.20  
1769        Electrónica      1363.45  
846         Electrónica      1483.20  
296          Decoración      1041.60  
2977        Electrónica       502.90  
34          Electrónica       435.25  
345          Decoración       299.13  
2370  Electrodomésticos       238.35  
527         Electrónica       252.16  
1847  Electrodomésticos      1181.00  
610         Electrónica       318.00  
2721  Electrodomésticos       230.74  
2529         Decoración       746.02  
2123  Electrodomésticos       485.76

In [59]:
# creo un nuevo DataFrame agrupando los productos y sumando los montos de las ventas
df_tabla_Ventas = df_ventas.groupby(["producto"], as_index=False)["monto_venta"].sum()

In [60]:
# corroboro la creación del nuevo DataFrame
df_tabla_Ventas.head()

producto  monto_venta
0  Adorno de pared     48093.49
1         Alfombra     44773.06
2       Aspiradora     50085.86
3      Auriculares     74175.58
4         Batidora     50979.20

In [61]:
# corroboro métricas estadísticas de la nueva tabla
df_tabla_Ventas.describe()

monto_venta
count     30.000000
mean   48903.117333
std    13265.561715
min    11128.800000
25%    45601.920000
50%    48140.245000
75%    51092.960000
max    82276.380000

In [62]:
# corroboro fórmula para obtener el Q3
df_tabla_Ventas['monto_venta'].quantile(0.75)

np.float64(51092.96)

In [63]:
# creo un nuevo DataFrame con Filtro los datos para quedarme con los productos de cuartil superior (Q3)
df_productosTop=df_tabla_Ventas[df_tabla_Ventas['monto_venta']>df_tabla_Ventas['monto_venta'].quantile(0.75)]

In [64]:
# corroboro el nuevo DataFrame con los producto más Vendidos
df_productosTop

producto  monto_venta
3         Auriculares     74175.58
5            Cafetera     59607.31
9   Cuadro decorativo     54297.60
16  Jarrón decorativo     51130.88
19    Lámpara de mesa     82276.38
20         Microondas     72562.89
25           Secadora     52115.45
27         Smartphone     54132.44

##***Actividad 3: Agregación***

 Resumir las ventas por categoría de producto y analizar los ingresos generados.

In [65]:
#creo un nuevo DataFrame agrupando los valores por categoría y sumando los montos de las ventas
df_ventas_Categoria = df_ventas.groupby(['categoria'], as_index=False)['monto_venta'].sum()

In [66]:
# corroboro la creación del nuevo DataFrame
df_ventas_Categoria

categoria  monto_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

In [67]:
# analizo el nuevo Dataframe y creo una nueva columna con el porcentaje de los montos de las ventas de cada categoría sobre el total
df_ventas_Categoria['porcentaje'] = df_ventas_Categoria['monto_venta']/df_ventas_Categoria['monto_venta'].sum()*100

In [68]:
# corroboro nueva columna
df_ventas_Categoria

categoria  monto_venta  porcentaje
0         Decoración    479216.09   32.664318
1  Electrodomésticos    505299.63   34.442224
2        Electrónica    482577.80   32.893459

##***Actividad 4: Integración de Datos***

Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.

###***Análisis de los campos de los DataFrame 'Ventas' y 'Marketing' para poder integrarlos***




In [69]:
# repaso las columnas del DataFrame Ventas
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id_venta     2998 non-null   int64         
 1   producto     2998 non-null   object        
 2   precio       2998 non-null   float64       
 3   cantidad     2998 non-null   float64       
 4   fecha_venta  2998 non-null   datetime64[ns]
 5   categoria    2998 non-null   object        
 6   monto_venta  2998 non-null   float64       
dtypes: datetime64[ns](1), float64(3), int64(1), object(2)
memory usage: 164.1+ KB


In [70]:
# repaso las columnas del DataFrame Marketin
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_campanha   90 non-null     int64         
 1   producto      90 non-null     object        
 2   canal         90 non-null     object        
 3   costo         90 non-null     float64       
 4   fecha_inicio  90 non-null     datetime64[ns]
 5   fecha_fin     90 non-null     datetime64[ns]
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 4.3+ KB


Ambos DataFrame comparten la categoría *'producto'* la cual puede ser utilizada para agrupar los DataFrame

###***Análisis del campo 'producto' en los DataFrame 'Ventas' y 'Marketing'***

In [71]:
# corroboro que los productos de ambos DataFrame sean los mismos
df_marketing['producto'].sort_values().unique() == df_ventas['producto'].sort_values().unique()

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True])

**Observación:**
- ambos DataFrame comparten los mismos productos.

####***DataFrame 'Marketing'***

In [72]:
# vemos cuantas veces se repite cada paroducto
df_marketing['producto'].sort_values().value_counts()

producto
Adorno de pared           3
Alfombra                  3
Aspiradora                3
Auriculares               3
Batidora                  3
Cafetera                  3
Candelabro                3
Consola de videojuegos    3
Cortinas                  3
Cuadro decorativo         3
Cámara digital            3
Elementos de cerámica     3
Espejo decorativo         3
Freidora eléctrica        3
Heladera                  3
Horno eléctrico           3
Jarrón decorativo         3
Laptop                    3
Lavadora                  3
Lámpara de mesa           3
Microondas                3
Parlantes Bluetooth       3
Plancha de vapor          3
Proyector                 3
Rincón de plantas         3
Secadora                  3
SmartWatch                3
Smartphone                3
Tablet                    3
Televisor                 3
Name: count, dtype: int64

In [73]:
# ordenamos los registros de "Marketing" en base a producto
df_marketing.sort_values(['producto']).head(30)

id_campanha                producto  canal  costo fecha_inicio  fecha_fin
0            74         Adorno de pared     TV   4.81   2024-03-20 2024-05-03
8            44         Adorno de pared  Email   5.08   2024-04-13 2024-05-10
70           14         Adorno de pared   RRSS   4.16   2024-10-22 2024-12-21
84           28                Alfombra   RRSS   5.82   2024-11-27 2025-01-08
4            58                Alfombra  Email   4.25   2024-03-31 2024-05-05
41           88                Alfombra     TV   7.39   2024-07-25 2024-08-31
85           70              Aspiradora     TV   3.06   2024-12-13 2024-12-29
29           40              Aspiradora  Email   6.24   2024-06-24 2024-08-12
52           10              Aspiradora   RRSS   7.16   2024-08-19 2024-09-19
24           47             Auriculares  Email   4.19   2024-06-02 2024-07-22
35           17             Auriculares   RRSS   6.06   2024-07-15 2024-08-15
79           77             Auriculares     TV   4.99   2024-11-11 2024-12-06
74           27                Batidora   RRSS   4.52   2024-11-01 2024-12-08
56           87                Batidora     TV   6.28   2024-08-27 2024-10-11
7            57                Batidora  Email   4.48   2024-04-10 2024-06-08
28           49                Cafetera  Email   5.20   2024-06-20 2024-07-22
53           19                Cafetera   RRSS   5.70   2024-08-19 2024-09-15
48           79                Cafetera     TV   7.26   2024-08-13 2024-10-11
68            9              Candelabro   RRSS   5.32   2024-10-08 2024-11-16
20           69              Candelabro     TV   3.25   2024-05-29 2024-06-30
38           39              Candelabro  Email   6.08   2024-07-17 2024-08-18
12           50  Consola de videojuegos  Email   4.97   2024-05-03 2024-05-28
36           20  Consola de videojuegos   RRSS   3.16   2024-07-15 2024-08-03
75           80  Consola de videojuegos     TV   4.64   2024-11-01 2024-11-20
11           43                Cortinas  Email   5.67   2024-05-02 2024-06-17
60           13                Cortinas   RRSS   5.22   2024-09-16 2024-10-15
51           73                Cortinas     TV   6.47   2024-08-17 2024-09-26
78           61       Cuadro decorativo     TV   5.30   2024-11-05 2024-12-23
10            1       Cuadro decorativo   RRSS   5.27   2024-04-27 2024-06-04
50           31       Cuadro decorativo  Email   5.28   2024-08-15 2024-09-12

**Observación:**

- hay 30 productos
- cada producto tiene 3 repeticiones correspondientes a cada tipo de canal

####***DataFrame 'Ventas'***

In [74]:
df_ventas['producto'].sort_values().value_counts()

producto
Lámpara de mesa           176
Auriculares               143
Microondas                135
Cafetera                  117
Rincón de plantas         101
Smartphone                101
Cuadro decorativo         100
Cortinas                  100
Espejo decorativo         100
Cámara digital            100
Alfombra                  100
Adorno de pared           100
Aspiradora                100
Batidora                  100
Televisor                 100
SmartWatch                100
Secadora                  100
Freidora eléctrica        100
Lavadora                  100
Laptop                    100
Horno eléctrico           100
Jarrón decorativo         100
Plancha de vapor          100
Proyector                  99
Consola de videojuegos     99
Elementos de cerámica      99
Parlantes Bluetooth        98
Tablet                     58
Heladera                   48
Candelabro                 24
Name: count, dtype: int64

####***Conclusión:***
- el campo *'productos'* puedes ser utilizado para integrar los datos de ambas tablas
- la organización de los datos de *'productos'* en ambos DataFrame es muy diferente:
    - en *'Marketing'* es regular, presentando 3 repeticiones para cada canal y se indica lo gastado en la  promoción del producto en un periodo de tiempo determinado.
      - Nº Registros: 90
    - en *'ventas'* es irregular, registrandose para cada  eventos de venta de estos el precio, la  cantidad y la fecha.
      - Nº Registros:2998
- La unión de los DataFrame sin modificaciones previas traerám repeticiones y mostrarán datos poco ejemplificadores de la realidad (precisos).

###***Integración de los DataFrame 'Ventas' y 'Marketing'***

In [75]:
# Integro los DataFrame 'Ventes' y 'Marketing' usando el campo 'producto'
ventas_marketing_1 = pd.merge(df_ventas,df_marketing, on='producto', how='inner')
ventas_marketing_1

id_venta                producto  precio  cantidad fecha_venta  \
0          792       Cuadro decorativo   69.94       5.0  2024-01-02   
1          792       Cuadro decorativo   69.94       5.0  2024-01-02   
2          792       Cuadro decorativo   69.94       5.0  2024-01-02   
3          811         Lámpara de mesa  105.10       5.0  2024-01-02   
4          811         Lámpara de mesa  105.10       5.0  2024-01-02   
...        ...                     ...     ...       ...         ...   
8989      2913              Smartphone   99.85       7.0  2024-12-30   
8990      2913              Smartphone   99.85       7.0  2024-12-30   
8991      2930  Consola de videojuegos   55.47       6.0  2024-12-30   
8992      2930  Consola de videojuegos   55.47       6.0  2024-12-30   
8993      2930  Consola de videojuegos   55.47       6.0  2024-12-30   

        categoria  monto_venta  id_campanha  canal  costo fecha_inicio  \
0      Decoración       349.70            1   RRSS   5.27   2024-04-27   
1      Decoración       349.70           31  Email   5.28   2024-08-15   
2      Decoración       349.70           61     TV   5.30   2024-11-05   
3      Decoración       525.50           32  Email   5.54   2024-03-28   
4      Decoración       525.50            2   RRSS   5.88   2024-05-30   
...           ...          ...          ...    ...    ...          ...   
8989  Electrónica       698.95           51  Email   4.65   2024-05-28   
8990  Electrónica       698.95           81     TV   4.78   2024-09-28   
8991  Electrónica       332.82           50  Email   4.97   2024-05-03   
8992  Electrónica       332.82           20   RRSS   3.16   2024-07-15   
8993  Electrónica       332.82           80     TV   4.64   2024-11-01   

      fecha_fin  
0    2024-06-04  
1    2024-09-12  
2    2024-12-23  
3    2024-04-20  
4    2024-06-29  
...         ...  
8989 2024-06-15  
8990 2024-10-24  
8991 2024-05-28  
8992 2024-08-03  
8993 2024-11-20  

[8994 rows x 12 columns]

####***Observaciones y Conclusiones***
**Observaciones:**
- integración de DataFrames poco útil ya que:
  - se repite cada registro de ventas para cada registro de marketing
  - se repiten cada registro de marketing para cada producto vendido.

**CONCLUSIÓN:**
- La integración de estas tablas en bruto además de ser poco informativa puede llevar a una muy mala interpretación de los datos.
- entre los posible usos que veo para la integración de los datos están:
  - compartar los gastos de marketing vs. las ganacias para cada producto
  - comparar las fechas de ventas y las de marketing para cada producto, para conocer si la estrategia de marketing fue efectiva
  - similar a lo anterior pero analizado la efectividad de los canales en las ventas de productos

###***Integración de los DataFrame 'Ventas' y 'Marketing' realizando transformaciones y agrupamientos de los datos***

**OBEJTIVO:**

> **Comparar los Gastos de Marketing de cada producto vs las ganancias obtenidas por sus ventas**

Para ello:

1. Crear un nuevo DataFrame en base a *'Marketing'* que para cada producto contenga:
  - la suma de los costos de marketing de los 3 canales.
    - **IMPORTANTE:** como las maginitudes de los costos de marketing estan en unidades, mientras que lor precios de *Ventas'* entas en decentas o cientos, **SE ASUME QUE LOS COSTOS ESTÁN EXPRESADOS EN MILES**.
  - como *'fecha_inicio'* y *'fecha_fin'* se tamará la fecha mínima y máxima de la campaña de cada producto. Se tomará el periodo completo de publicidad independientemente del canal.
2. Crear un nuevo DataFrame en base a *'Ventas'* que para cada producto contenga:
  - la categoría
  - la suma de los montos de las ventas

####***Nuevos DataFrame Marketing Producto***

In [76]:
# creo un nuevo DataFrame a partir de Marketing
df_MarketingProductos = df_marketing.groupby(['producto'], as_index=False).agg({
    "costo": "sum",
    "fecha_inicio": "min",
    "fecha_fin": "max"
})

In [77]:
# corrorobo nuevo DataFrame a partir de Ventas de Ventas
df_MarketingProductos

producto  costo fecha_inicio  fecha_fin
0          Adorno de pared  14.05   2024-03-20 2024-12-21
1                 Alfombra  17.46   2024-03-31 2025-01-08
2               Aspiradora  16.46   2024-06-24 2024-12-29
3              Auriculares  15.24   2024-06-02 2024-12-06
4                 Batidora  15.28   2024-04-10 2024-12-08
5                 Cafetera  18.16   2024-06-20 2024-10-11
6               Candelabro  14.65   2024-05-29 2024-11-16
7   Consola de videojuegos  12.77   2024-05-03 2024-11-20
8                 Cortinas  17.36   2024-05-02 2024-10-15
9        Cuadro decorativo  15.85   2024-04-27 2024-12-23
10          Cámara digital  16.40   2024-07-16 2024-12-09
11   Elementos de cerámica  16.87   2024-06-26 2024-10-12
12       Espejo decorativo  14.89   2024-05-31 2024-10-17
13      Freidora eléctrica  13.56   2024-06-05 2025-01-21
14                Heladera  13.11   2024-05-06 2024-08-09
15         Horno eléctrico  13.89   2024-07-31 2024-12-11
16       Jarrón decorativo  13.27   2024-05-09 2024-12-25
17                  Laptop  15.19   2024-09-28 2025-01-13
18                Lavadora  11.85   2024-05-12 2024-09-20
19         Lámpara de mesa  15.93   2024-03-28 2024-10-07
20              Microondas  14.19   2024-08-06 2024-11-15
21     Parlantes Bluetooth  13.37   2024-04-17 2024-10-26
22        Plancha de vapor  14.12   2024-04-02 2024-11-02
23               Proyector  12.47   2024-05-22 2024-08-16
24       Rincón de plantas  16.45   2024-06-14 2025-02-14
25                Secadora  14.59   2024-07-24 2025-01-07
26              SmartWatch  14.50   2024-04-01 2024-12-29
27              Smartphone  15.80   2024-03-29 2024-10-24
28                  Tablet  12.72   2024-03-26 2024-09-03
29               Televisor  13.13   2024-05-17 2025-02-08

In [78]:
# Según lo asumido de que los costos están expresados en miles, multiplico los costos por mil
df_MarketingProductos['costo'] = df_MarketingProductos['costo']*1000

In [79]:
# corroboro la modificación de 'costos'
df_MarketingProductos

producto    costo fecha_inicio  fecha_fin
0          Adorno de pared  14050.0   2024-03-20 2024-12-21
1                 Alfombra  17460.0   2024-03-31 2025-01-08
2               Aspiradora  16460.0   2024-06-24 2024-12-29
3              Auriculares  15240.0   2024-06-02 2024-12-06
4                 Batidora  15280.0   2024-04-10 2024-12-08
5                 Cafetera  18160.0   2024-06-20 2024-10-11
6               Candelabro  14650.0   2024-05-29 2024-11-16
7   Consola de videojuegos  12770.0   2024-05-03 2024-11-20
8                 Cortinas  17360.0   2024-05-02 2024-10-15
9        Cuadro decorativo  15850.0   2024-04-27 2024-12-23
10          Cámara digital  16400.0   2024-07-16 2024-12-09
11   Elementos de cerámica  16870.0   2024-06-26 2024-10-12
12       Espejo decorativo  14890.0   2024-05-31 2024-10-17
13      Freidora eléctrica  13560.0   2024-06-05 2025-01-21
14                Heladera  13110.0   2024-05-06 2024-08-09
15         Horno eléctrico  13890.0   2024-07-31 2024-12-11
16       Jarrón decorativo  13270.0   2024-05-09 2024-12-25
17                  Laptop  15190.0   2024-09-28 2025-01-13
18                Lavadora  11850.0   2024-05-12 2024-09-20
19         Lámpara de mesa  15930.0   2024-03-28 2024-10-07
20              Microondas  14190.0   2024-08-06 2024-11-15
21     Parlantes Bluetooth  13370.0   2024-04-17 2024-10-26
22        Plancha de vapor  14120.0   2024-04-02 2024-11-02
23               Proyector  12470.0   2024-05-22 2024-08-16
24       Rincón de plantas  16450.0   2024-06-14 2025-02-14
25                Secadora  14590.0   2024-07-24 2025-01-07
26              SmartWatch  14500.0   2024-04-01 2024-12-29
27              Smartphone  15800.0   2024-03-29 2024-10-24
28                  Tablet  12720.0   2024-03-26 2024-09-03
29               Televisor  13130.0   2024-05-17 2025-02-08

####***Nuevos DataFrame Venta Producto***

In [80]:
# creo un nuevo DataFrame a partir de Ventas
df_VentaProductos = df_ventas.groupby(['categoria', 'producto'], as_index=False)["monto_venta"].sum()

In [81]:
# corrorobo nuevo DataFrame a partir de Ventas de Ventas
df_VentaProductos

categoria                producto  monto_venta
0          Decoración         Adorno de pared     48093.49
1          Decoración                Alfombra     44773.06
2          Decoración              Candelabro     11128.80
3          Decoración                Cortinas     44865.03
4          Decoración       Cuadro decorativo     54297.60
5          Decoración   Elementos de cerámica     45411.09
6          Decoración       Espejo decorativo     46783.31
7          Decoración       Jarrón decorativo     51130.88
8          Decoración         Lámpara de mesa     82276.38
9          Decoración       Rincón de plantas     50456.45
10  Electrodomésticos              Aspiradora     50085.86
11  Electrodomésticos                Batidora     50979.20
12  Electrodomésticos                Cafetera     59607.31
13  Electrodomésticos      Freidora eléctrica     50155.15
14  Electrodomésticos                Heladera     25736.80
15  Electrodomésticos         Horno eléctrico     49913.90
16  Electrodomésticos                Lavadora     48946.44
17  Electrodomésticos              Microondas     72562.89
18  Electrodomésticos        Plancha de vapor     45196.63
19  Electrodomésticos                Secadora     52115.45
20        Electrónica             Auriculares     74175.58
21        Electrónica  Consola de videojuegos     46174.41
22        Electrónica          Cámara digital     45217.96
23        Electrónica                  Laptop     46179.48
24        Electrónica     Parlantes Bluetooth     47779.46
25        Electrónica               Proyector     48187.00
26        Electrónica              SmartWatch     47836.86
27        Electrónica              Smartphone     54132.44
28        Electrónica                  Tablet     25856.57
29        Electrónica               Televisor     47038.04

####***Integro los nuevos DataFrames***

In [82]:
# integración de los DataFrames 'df_MarketingProductos' y 'df_VentaProductos'
df_productos = pd.merge(df_VentaProductos, df_MarketingProductos, on='producto', how='inner')

In [83]:
# corroboro nuevo DataFrame
df_productos

categoria                producto  monto_venta    costo  \
0          Decoración         Adorno de pared     48093.49  14050.0   
1          Decoración                Alfombra     44773.06  17460.0   
2          Decoración              Candelabro     11128.80  14650.0   
3          Decoración                Cortinas     44865.03  17360.0   
4          Decoración       Cuadro decorativo     54297.60  15850.0   
5          Decoración   Elementos de cerámica     45411.09  16870.0   
6          Decoración       Espejo decorativo     46783.31  14890.0   
7          Decoración       Jarrón decorativo     51130.88  13270.0   
8          Decoración         Lámpara de mesa     82276.38  15930.0   
9          Decoración       Rincón de plantas     50456.45  16450.0   
10  Electrodomésticos              Aspiradora     50085.86  16460.0   
11  Electrodomésticos                Batidora     50979.20  15280.0   
12  Electrodomésticos                Cafetera     59607.31  18160.0   
13  Electrodomésticos      Freidora eléctrica     50155.15  13560.0   
14  Electrodomésticos                Heladera     25736.80  13110.0   
15  Electrodomésticos         Horno eléctrico     49913.90  13890.0   
16  Electrodomésticos                Lavadora     48946.44  11850.0   
17  Electrodomésticos              Microondas     72562.89  14190.0   
18  Electrodomésticos        Plancha de vapor     45196.63  14120.0   
19  Electrodomésticos                Secadora     52115.45  14590.0   
20        Electrónica             Auriculares     74175.58  15240.0   
21        Electrónica  Consola de videojuegos     46174.41  12770.0   
22        Electrónica          Cámara digital     45217.96  16400.0   
23        Electrónica                  Laptop     46179.48  15190.0   
24        Electrónica     Parlantes Bluetooth     47779.46  13370.0   
25        Electrónica               Proyector     48187.00  12470.0   
26        Electrónica              SmartWatch     47836.86  14500.0   
27        Electrónica              Smartphone     54132.44  15800.0   
28        Electrónica                  Tablet     25856.57  12720.0   
29        Electrónica               Televisor     47038.04  13130.0   

   fecha_inicio  fecha_fin  
0    2024-03-20 2024-12-21  
1    2024-03-31 2025-01-08  
2    2024-05-29 2024-11-16  
3    2024-05-02 2024-10-15  
4    2024-04-27 2024-12-23  
5    2024-06-26 2024-10-12  
6    2024-05-31 2024-10-17  
7    2024-05-09 2024-12-25  
8    2024-03-28 2024-10-07  
9    2024-06-14 2025-02-14  
10   2024-06-24 2024-12-29  
11   2024-04-10 2024-12-08  
12   2024-06-20 2024-10-11  
13   2024-06-05 2025-01-21  
14   2024-05-06 2024-08-09  
15   2024-07-31 2024-12-11  
16   2024-05-12 2024-09-20  
17   2024-08-06 2024-11-15  
18   2024-04-02 2024-11-02  
19   2024-07-24 2025-01-07  
20   2024-06-02 2024-12-06  
21   2024-05-03 2024-11-20  
22   2024-07-16 2024-12-09  
23   2024-09-28 2025-01-13  
24   2024-04-17 2024-10-26  
25   2024-05-22 2024-08-16  
26   2024-04-01 2024-12-29  
27   2024-03-29 2024-10-24  
28   2024-03-26 2024-09-03  
29   2024-05-17 2025-02-08

####***Análisis de Ganancias en nuevo DataFrame Integrado***

In [84]:
# agrego la columna ganancias como 'montos_venta' - 'costo'
df_productos['ganancia'] = df_productos['monto_venta'] - df_productos['costo']

In [85]:
# corroboro nueva columna
df_productos

categoria                producto  monto_venta    costo  \
0          Decoración         Adorno de pared     48093.49  14050.0   
1          Decoración                Alfombra     44773.06  17460.0   
2          Decoración              Candelabro     11128.80  14650.0   
3          Decoración                Cortinas     44865.03  17360.0   
4          Decoración       Cuadro decorativo     54297.60  15850.0   
5          Decoración   Elementos de cerámica     45411.09  16870.0   
6          Decoración       Espejo decorativo     46783.31  14890.0   
7          Decoración       Jarrón decorativo     51130.88  13270.0   
8          Decoración         Lámpara de mesa     82276.38  15930.0   
9          Decoración       Rincón de plantas     50456.45  16450.0   
10  Electrodomésticos              Aspiradora     50085.86  16460.0   
11  Electrodomésticos                Batidora     50979.20  15280.0   
12  Electrodomésticos                Cafetera     59607.31  18160.0   
13  Electrodomésticos      Freidora eléctrica     50155.15  13560.0   
14  Electrodomésticos                Heladera     25736.80  13110.0   
15  Electrodomésticos         Horno eléctrico     49913.90  13890.0   
16  Electrodomésticos                Lavadora     48946.44  11850.0   
17  Electrodomésticos              Microondas     72562.89  14190.0   
18  Electrodomésticos        Plancha de vapor     45196.63  14120.0   
19  Electrodomésticos                Secadora     52115.45  14590.0   
20        Electrónica             Auriculares     74175.58  15240.0   
21        Electrónica  Consola de videojuegos     46174.41  12770.0   
22        Electrónica          Cámara digital     45217.96  16400.0   
23        Electrónica                  Laptop     46179.48  15190.0   
24        Electrónica     Parlantes Bluetooth     47779.46  13370.0   
25        Electrónica               Proyector     48187.00  12470.0   
26        Electrónica              SmartWatch     47836.86  14500.0   
27        Electrónica              Smartphone     54132.44  15800.0   
28        Electrónica                  Tablet     25856.57  12720.0   
29        Electrónica               Televisor     47038.04  13130.0   

   fecha_inicio  fecha_fin  ganancia  
0    2024-03-20 2024-12-21  34043.49  
1    2024-03-31 2025-01-08  27313.06  
2    2024-05-29 2024-11-16  -3521.20  
3    2024-05-02 2024-10-15  27505.03  
4    2024-04-27 2024-12-23  38447.60  
5    2024-06-26 2024-10-12  28541.09  
6    2024-05-31 2024-10-17  31893.31  
7    2024-05-09 2024-12-25  37860.88  
8    2024-03-28 2024-10-07  66346.38  
9    2024-06-14 2025-02-14  34006.45  
10   2024-06-24 2024-12-29  33625.86  
11   2024-04-10 2024-12-08  35699.20  
12   2024-06-20 2024-10-11  41447.31  
13   2024-06-05 2025-01-21  36595.15  
14   2024-05-06 2024-08-09  12626.80  
15   2024-07-31 2024-12-11  36023.90  
16   2024-05-12 2024-09-20  37096.44  
17   2024-08-06 2024-11-15  58372.89  
18   2024-04-02 2024-11-02  31076.63  
19   2024-07-24 2025-01-07  37525.45  
20   2024-06-02 2024-12-06  58935.58  
21   2024-05-03 2024-11-20  33404.41  
22   2024-07-16 2024-12-09  28817.96  
23   2024-09-28 2025-01-13  30989.48  
24   2024-04-17 2024-10-26  34409.46  
25   2024-05-22 2024-08-16  35717.00  
26   2024-04-01 2024-12-29  33336.86  
27   2024-03-29 2024-10-24  38332.44  
28   2024-03-26 2024-09-03  13136.57  
29   2024-05-17 2025-02-08  33908.04

In [86]:
df_productos.sort_values('ganancia')

categoria                producto  monto_venta    costo  \
2          Decoración              Candelabro     11128.80  14650.0   
14  Electrodomésticos                Heladera     25736.80  13110.0   
28        Electrónica                  Tablet     25856.57  12720.0   
1          Decoración                Alfombra     44773.06  17460.0   
3          Decoración                Cortinas     44865.03  17360.0   
5          Decoración   Elementos de cerámica     45411.09  16870.0   
22        Electrónica          Cámara digital     45217.96  16400.0   
23        Electrónica                  Laptop     46179.48  15190.0   
18  Electrodomésticos        Plancha de vapor     45196.63  14120.0   
6          Decoración       Espejo decorativo     46783.31  14890.0   
26        Electrónica              SmartWatch     47836.86  14500.0   
21        Electrónica  Consola de videojuegos     46174.41  12770.0   
10  Electrodomésticos              Aspiradora     50085.86  16460.0   
29        Electrónica               Televisor     47038.04  13130.0   
9          Decoración       Rincón de plantas     50456.45  16450.0   
0          Decoración         Adorno de pared     48093.49  14050.0   
24        Electrónica     Parlantes Bluetooth     47779.46  13370.0   
11  Electrodomésticos                Batidora     50979.20  15280.0   
25        Electrónica               Proyector     48187.00  12470.0   
15  Electrodomésticos         Horno eléctrico     49913.90  13890.0   
13  Electrodomésticos      Freidora eléctrica     50155.15  13560.0   
16  Electrodomésticos                Lavadora     48946.44  11850.0   
19  Electrodomésticos                Secadora     52115.45  14590.0   
7          Decoración       Jarrón decorativo     51130.88  13270.0   
27        Electrónica              Smartphone     54132.44  15800.0   
4          Decoración       Cuadro decorativo     54297.60  15850.0   
12  Electrodomésticos                Cafetera     59607.31  18160.0   
17  Electrodomésticos              Microondas     72562.89  14190.0   
20        Electrónica             Auriculares     74175.58  15240.0   
8          Decoración         Lámpara de mesa     82276.38  15930.0   

   fecha_inicio  fecha_fin  ganancia  
2    2024-05-29 2024-11-16  -3521.20  
14   2024-05-06 2024-08-09  12626.80  
28   2024-03-26 2024-09-03  13136.57  
1    2024-03-31 2025-01-08  27313.06  
3    2024-05-02 2024-10-15  27505.03  
5    2024-06-26 2024-10-12  28541.09  
22   2024-07-16 2024-12-09  28817.96  
23   2024-09-28 2025-01-13  30989.48  
18   2024-04-02 2024-11-02  31076.63  
6    2024-05-31 2024-10-17  31893.31  
26   2024-04-01 2024-12-29  33336.86  
21   2024-05-03 2024-11-20  33404.41  
10   2024-06-24 2024-12-29  33625.86  
29   2024-05-17 2025-02-08  33908.04  
9    2024-06-14 2025-02-14  34006.45  
0    2024-03-20 2024-12-21  34043.49  
24   2024-04-17 2024-10-26  34409.46  
11   2024-04-10 2024-12-08  35699.20  
25   2024-05-22 2024-08-16  35717.00  
15   2024-07-31 2024-12-11  36023.90  
13   2024-06-05 2025-01-21  36595.15  
16   2024-05-12 2024-09-20  37096.44  
19   2024-07-24 2025-01-07  37525.45  
7    2024-05-09 2024-12-25  37860.88  
27   2024-03-29 2024-10-24  38332.44  
4    2024-04-27 2024-12-23  38447.60  
12   2024-06-20 2024-10-11  41447.31  
17   2024-08-06 2024-11-15  58372.89  
20   2024-06-02 2024-12-06  58935.58  
8    2024-03-28 2024-10-07  66346.38

In [87]:
df_productos.groupby(['categoria'])['ganancia'].sum()

categoria
Decoración           322436.09
Electrodomésticos    360089.63
Electrónica          340987.80
Name: ganancia, dtype: float64

**Observación:**
- el producto que más ganancia dio fueron las *'Lamparas de Mesa' (Decoración)*.
- Los *'Candelabros' (Decoración)* fueron los únicos productos que dieron pérdida.
- La categoría que más ganancia dió fue *'Electrodomésticos'*

#**Etapa 3: Análisis de Datos**

In [88]:
# Para recordar todos los DataFrame utilizads hasta ahora
%who DataFrame

df_MarketingProductos	 df_VentaProductos	 df_clientes	 df_marketing	 df_productos	 df_productosTop	 df_tabla_Ventas	 df_ventas	 df_ventas_Categoria	 
ventas_marketing_1	 


##***Actividad 1: Estadística Descriptiva***

### ***DataFrame Ventas***

#### ***Modificación del DataFrame Ventas***

Al DataFrame 'Ventas' se le agregaran los campos "mes" y "día_Semana" para realizar los análisis estadisticos en base a los meses y los días de la semana

In [89]:
# Agrego el campo "mes" al DataFrame 'Ventas'
df_ventas['mes'] = df_ventas['fecha_venta'].dt.month

# Agrego el campo "dia_semana" al DataFrame 'Ventas'
df_ventas['dia_semana'] = df_ventas['fecha_venta'].dt.day_name()

# Corroboro que se realizaron los cambios correctamente
df_ventas

id_venta                producto  precio  cantidad fecha_venta  \
0          792       Cuadro decorativo   69.94       5.0  2024-01-02   
1          811         Lámpara de mesa  105.10       5.0  2024-01-02   
2         1156                Secadora   97.96       3.0  2024-01-02   
3         1372                Heladera  114.35       8.0  2024-01-02   
4         1546                Secadora  106.21       4.0  2024-01-02   
...        ...                     ...     ...       ...         ...   
2993      1837         Horno eléctrico  104.12       9.0  2024-12-30   
2994      2276                  Laptop   85.27       9.0  2024-12-30   
2995      2696                  Laptop  107.81       4.0  2024-12-30   
2996      2913              Smartphone   99.85       7.0  2024-12-30   
2997      2930  Consola de videojuegos   55.47       6.0  2024-12-30   

              categoria  monto_venta  mes dia_semana  
0            Decoración       349.70    1    Tuesday  
1            Decoración       525.50    1    Tuesday  
2     Electrodomésticos       293.88    1    Tuesday  
3     Electrodomésticos       914.80    1    Tuesday  
4     Electrodomésticos       424.84    1    Tuesday  
...                 ...          ...  ...        ...  
2993  Electrodomésticos       937.08   12     Monday  
2994        Electrónica       767.43   12     Monday  
2995        Electrónica       431.24   12     Monday  
2996        Electrónica       698.95   12     Monday  
2997        Electrónica       332.82   12     Monday  

[2998 rows x 9 columns]

#### ***Estadísticas del DataFrame Ventas***

In [96]:
# Estadísticas generales del DataFrame Ventas
df_ventas[["precio", "cantidad", "monto_venta"]].describe()



precio     cantidad  monto_venta
count  2998.000000  2998.000000  2998.000000
mean     75.291318     6.502668   489.357412
std      28.736008     3.455107   334.281817
min      26.000000     1.000000    26.300000
25%      50.030000     3.000000   220.920000
50%      75.205000     7.000000   418.065000
75%     100.067500     9.000000   709.920000
max     124.970000    12.000000  1488.120000

#####***Estadísticas de Ventas por mes***

In [99]:
# Estadisticas del campo 'precio' agrupadas por mes
df_ventas.groupby("mes")[["precio"]].describe()


precio                                                                
     count       mean        std    min      25%     50%       75%     max
mes                                                                       
1    268.0  74.990075  29.726902  26.42  47.7425  78.910   99.7000  123.96
2    248.0  74.913790  28.744975  26.00  49.2300  74.645   99.9450  124.82
3    265.0  75.769547  28.922329  27.65  49.3700  76.150  101.5700  124.82
4    275.0  74.537055  29.726053  26.29  46.4850  73.700   98.2500  124.95
5    275.0  77.984982  28.949767  26.06  52.0850  78.440  104.5050  124.57
6    221.0  76.176516  28.471586  26.82  51.5100  74.810  103.1900  124.91
7    243.0  74.831770  27.301862  26.30  52.6200  74.880   97.1500  124.64
8    254.0  75.696654  27.595050  26.11  53.3325  75.965   96.5450  124.96
9    241.0  73.831826  28.924087  26.03  50.2000  71.400   97.1700  124.97
10   230.0  77.531435  29.486885  26.32  52.2675  77.105  104.0975  124.48
11   241.0  73.180830  28.437006  27.72  49.4700  70.550   98.1500  124.49
12   237.0  73.909536  28.455116  27.21  47.7700  75.000   98.5800  124.24

In [98]:
# Estadisticas del campo 'cantidad' agrupadas por mes
df_ventas.groupby("mes")[["cantidad"]].describe()


cantidad                                                
       count      mean       std  min   25%  50%   75%   max
mes                                                         
1      268.0  6.496269  3.465721  1.0  3.75  7.0   9.0  12.0
2      248.0  6.479839  3.464043  1.0  4.00  6.0  10.0  12.0
3      265.0  6.762264  3.672184  1.0  3.00  7.0  10.0  12.0
4      275.0  6.236364  3.482824  1.0  3.00  7.0   9.0  12.0
5      275.0  6.589091  3.431588  1.0  4.00  6.0  10.0  12.0
6      221.0  6.452489  3.478996  1.0  3.00  6.0   9.0  12.0
7      243.0  6.366255  3.585095  1.0  3.00  6.0   9.0  12.0
8      254.0  6.354331  3.466446  1.0  3.00  6.0   9.0  12.0
9      241.0  6.539419  3.247997  1.0  4.00  6.0   9.0  12.0
10     230.0  6.339130  3.482676  1.0  4.00  6.0  10.0  12.0
11     241.0  6.742739  3.450875  1.0  4.00  7.0  10.0  12.0
12     237.0  6.675105  3.214100  1.0  4.00  7.0   9.0  12.0

In [100]:
# Estadisticas del campo 'monto_ventas' agrupadas por mes
df_ventas.groupby("mes")[["monto_venta"]].describe()

monto_venta                                                              \
          count        mean         std    min       25%      50%       75%   
mes                                                                           
1         268.0  483.600709  340.244520  29.17  220.3850  388.600  689.7900   
2         248.0  478.517903  316.761425  31.09  225.9700  426.475  695.1600   
3         265.0  516.147736  360.610738  31.42  224.8500  444.430  761.0400   
4         275.0  467.020691  344.828281  29.68  184.1700  368.400  653.6850   
5         275.0  522.644545  352.553560  29.73  221.4500  456.800  789.0300   
6         221.0  490.860498  328.492216  33.73  210.8800  415.640  742.2100   
7         243.0  478.312634  333.361886  26.30  218.7750  420.480  682.8850   
8         254.0  471.181693  320.801691  36.21  221.1575  394.260  683.2850   
9         241.0  480.447510  321.042204  34.56  237.7800  421.440  660.0600   
10        230.0  487.465783  335.691883  38.69  209.2875  387.585  723.4325   
11        241.0  497.725270  340.307145  34.70  229.8600  413.900  722.9400   
12        237.0  496.337300  308.070113  27.83  254.4900  449.160  735.4800   

              
         max  
mes           
1    1461.24  
2    1464.24  
3    1488.12  
4    1485.96  
5    1367.74  
6    1440.24  
7    1412.16  
8    1404.84  
9    1449.48  
10   1480.80  
11   1458.72  
12   1308.84

#####***Estadísticas de Ventas por día de la semana***

In [102]:
# Estadisticas del campo 'precio' agrupadas por día de la semana
df_ventas.groupby("dia_semana")[["precio"]].describe()

precio                                                          \
            count       mean        std    min      25%     50%       75%   
dia_semana                                                                  
Friday      378.0  74.282143  28.370600  26.00  50.7025  73.795   97.9100   
Monday      418.0  76.357799  28.346499  26.24  50.8975  77.195  100.2550   
Saturday    444.0  74.836261  28.399973  26.11  51.3575  73.415   98.2325   
Sunday      447.0  76.615436  28.649518  26.06  51.6250  77.130  101.9300   
Thursday    435.0  75.523149  28.806071  26.29  49.9300  76.040   99.6550   
Tuesday     439.0  73.164897  28.743804  26.32  47.7700  69.940   97.7000   
Wednesday   437.0  76.157437  29.786111  26.20  49.0600  78.130  101.5800   

                    
               max  
dia_semana          
Friday      124.97  
Monday      124.95  
Saturday    124.57  
Sunday      124.83  
Thursday    124.96  
Tuesday     124.91  
Wednesday   124.88

In [103]:
# Estadisticas del campo 'cantidad' agrupadas por día de la semana
df_ventas.groupby("dia_semana")[["cantidad"]].describe()

cantidad                                               
              count      mean       std  min  25%  50%   75%   max
dia_semana                                                        
Friday        378.0  6.785714  3.437059  1.0  4.0  7.0  10.0  12.0
Monday        418.0  6.208134  3.517306  1.0  3.0  6.0   9.0  12.0
Saturday      444.0  6.531532  3.354374  1.0  4.0  7.0   9.0  12.0
Sunday        447.0  6.463087  3.396497  1.0  3.0  6.0   9.0  12.0
Thursday      435.0  6.786207  3.532312  1.0  4.0  7.0  10.0  12.0
Tuesday       439.0  6.195900  3.446642  1.0  3.0  6.0   9.0  12.0
Wednesday     437.0  6.576659  3.476884  1.0  4.0  7.0  10.0  12.0

In [104]:
# Estadisticas del campo 'monto_venta' agrupadas por día de la semana
df_ventas.groupby("dia_semana")[["monto_venta"]].describe()

monto_venta                                                    \
                 count        mean         std    min       25%      50%   
dia_semana                                                                 
Friday           378.0  506.526323  343.185778  28.91  233.6400  427.825   
Monday           418.0  476.207440  336.714459  29.17  184.2675  391.935   
Saturday         444.0  484.274550  322.534300  32.30  227.7400  422.190   
Sunday           447.0  491.545347  324.844616  31.42  234.5600  421.680   
Thursday         435.0  507.151149  336.120710  29.76  235.3850  443.520   
Tuesday          439.0  457.277950  330.222794  30.59  189.0300  374.730   
Wednesday        437.0  504.524989  346.634023  26.30  235.4200  418.330   

                               
                 75%      max  
dia_semana                     
Friday      723.1275  1458.72  
Monday      728.1300  1452.00  
Saturday    699.6825  1422.60  
Sunday      712.7800  1483.20  
Thursday    722.6850  1485.96  
Tuesday     665.3800  1488.12  
Wednesday   723.8000  1461.24

### ***DataFrame Marketing***

In [107]:
# Estadisticas del campo 'costos'
df_marketing [["costo"]].describe()

costo
count  90.000000
mean    4.928667
std     0.947750
min     2.950000
25%     4.372500
50%     4.900000
75%     5.562500
max     7.390000

In [106]:
# Estadisticas del campo 'costos' agrupadas por canal
df_marketing.groupby("canal")[["costo"]].describe()

costo                                                       
      count      mean       std   min     25%    50%     75%   max
canal                                                             
Email  30.0  4.840000  0.898773  2.95  4.2125  4.895  5.5075  6.84
RRSS   30.0  5.030333  0.894101  3.16  4.5600  5.130  5.5425  7.16
TV     30.0  4.915667  1.063215  3.06  4.3725  4.795  5.5400  7.39

##***Actividad 2: Análisis Exploratorio de Datos (EDA)***

### ***DataFrame Ventas***

realizar las descripciones de boxplot por semana y mes... cantidad de ventas o montos de ventas por producto por mes.

se pueden personalizar los gráficos es esta misma etapa

### ***DataFrame Marketing***

##***Actividad 3: Correlación***

##***Actividad 4: Visualización Avanzada con Seaborn***

#**Etapa 4: Consolidación de Datos**

##***Actividad 1: Visualización con Matplotlib***

##***Actividad 2: Visualización Avanzada con Seaborn***

##***Actividad 3: Visualización Interactiva con Plotly***

##***Actividad 4: Presentación Final***

#**Anexo**

**Bases de datos utilizadas:**
- [clientes.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/clientes.csv)
- [marketing.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/marketing.csv)
- [ventas.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/ventas.csv)

In [90]:
ventas_marketing_1.head()


id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94       5.0  2024-01-02  Decoración   
1       792  Cuadro decorativo   69.94       5.0  2024-01-02  Decoración   
2       792  Cuadro decorativo   69.94       5.0  2024-01-02  Decoración   
3       811    Lámpara de mesa  105.10       5.0  2024-01-02  Decoración   
4       811    Lámpara de mesa  105.10       5.0  2024-01-02  Decoración   

   monto_venta  id_campanha  canal  costo fecha_inicio  fecha_fin  
0        349.7            1   RRSS   5.27   2024-04-27 2024-06-04  
1        349.7           31  Email   5.28   2024-08-15 2024-09-12  
2        349.7           61     TV   5.30   2024-11-05 2024-12-23  
3        525.5           32  Email   5.54   2024-03-28 2024-04-20  
4        525.5            2   RRSS   5.88   2024-05-30 2024-06-29